atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/QosCassandraClient.java

```diff
@@ -118,56 +115,33 @@ private long getApproximateWriteByteCount(Map<ByteBuffer, Map<String, List<Mutat
     public ColumnOrSuperColumn get(TableReference tableReference, ByteBuffer key, byte[] column,
             ConsistencyLevel consistency_level)
             throws InvalidRequestException, NotFoundException, UnavailableException, TimedOutException, TException {
-        qosClient.checkLimit();
-
-        ColumnOrSuperColumn result = client.get(tableReference, key, column, consistency_level);
-        recordBytesRead(() -> ThriftObjectSizeUtils.getColumnOrSuperColumnSize(result));
-        return result;
+        return qosClient.executeRead(
+                () -> DEFAULT_ESTIMATED_READ_BYTES,
+                () -> client.get(tableReference, key, column, consistency_level),
+                ThriftObjectSizeUtils::getColumnOrSuperColumnSize);
     }
 
     @Override
     public CASResult cas(TableReference tableReference, ByteBuffer key, List<Column> expected, List<Column> updates,
             ConsistencyLevel serial_consistency_level, ConsistencyLevel commit_consistency_level)
             throws InvalidRequestException, UnavailableException, TimedOutException, TException {
-        qosClient.checkLimit();
-
-        CASResult result = client.cas(tableReference, key, expected, updates, serial_consistency_level,
+        // TODO(nziebart): should this be considered as a write or do we need to treat is as both read and write?
```

Comment:
Sounds likely - I would expect that CAS operations are twice as heavy for C* as reads/writes.